server/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -735,6 +736,10 @@ public Object readGenericValue() throws IOException {
                 return readGeoPoint();
             case 23:
                 return readZonedDateTime();
+            case 24:
+                return readCollection(StreamInput::readGenericValue, LinkedHashSet::new, Collections.emptySet());
+            case 25:
+                return readCollection(StreamInput::readGenericValue, HashSet::new, Collections.emptySet());
```

Comment:
I think you can just make this `readSet(StreamInput::readGenericValue)`?

Reply:
It could work for the HashSet version, but I think it is good to have the symmetry with with the LInkedHashSet above.

Reply:
fair point :) LGTM